SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityJobPoller.java

```diff
@@ -76,13 +81,34 @@ public void runActionOnPoll() {
     final Map<String, SingularityRequestWithState> idToRequest = Maps.uniqueIndex(requests, SingularityRequestWithState.REQUEST_STATE_TO_REQUEST_ID);
 
     for (SingularityTaskId taskId : activeTaskIds) {
-      SingularityRequestWithState request = idToRequest.get(taskId.getRequestId());
+      SingularityRequestWithState requestWithState = idToRequest.get(taskId.getRequestId());
 
-      if (request == null || !request.getRequest().isScheduled() || taskManager.hasNotifiedOverdue(taskId)) {
+      if (requestWithState == null) {
         continue;
       }
 
+      SingularityRequest request = requestWithState.getRequest();
       final long runtime = start - taskId.getStartedAt();
+
+      if (!request.getRequestType().isLongRunning() &&
+          request.getTaskExecutionTimeLimitMillis().or(configuration.getTaskExecutionTimeLimitMillis()).isPresent() &&
+          runtime >= request.getTaskExecutionTimeLimitMillis().or(configuration.getTaskExecutionTimeLimitMillis().get())) {
+
+        taskManager.createTaskCleanup(new SingularityTaskCleanup(
+            Optional.<String>absent(),
+            TaskCleanupType.TASK_EXCEEDED_TIME_LIMIT,
+            start,
+            taskId,
+            Optional.of(String.format("Task has run for %s, which exceeds the maximum execution time of %s",
+                DurationFormatUtils.formatDurationHMS(runtime),
+                DurationFormatUtils.formatDurationHMS(configuration.getTaskExecutionTimeLimitMillis().get()))),
```

Comment:
Just an informational statement, but will want the request.get...().or(config.... here as well for accuracy